src/javascript/app_2/App/Components/Form/Datepicker/datepicker.jsx

```diff
@@ -0,0 +1,202 @@
+import classNames      from 'classnames';
+import moment          from 'moment';
+import React           from 'react';
+import DatePickerInput from './datepicker_input.jsx';
+import ArrowHead       from '../../Elements/arrowhead.jsx';
+import Calendar        from '../../Elements/Calendar';
+import {
+    formatDate,
+    daysFromTodayTo }  from '../../../../Utils/Date';
+
+class DatePicker extends React.PureComponent {
+    state = {
+        value                : '',
+        is_datepicker_visible: false,
+        is_close_btn_visible : false,
+    };
+
+    componentWillReceiveProps({ value, mode }) {
```

Comment:
@kellybinary 
a deprecated lifecycle method.

Reply:
@kellybinary You can use `componentDidUpdate()` instead.

Reference: [link](https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops)